OpenSimRT/Common/include/CircularBuffer.h

```diff
@@ -21,34 +21,99 @@
  *
  * \brief Implementation of a thread safe circular buffer.
  *
- * @author Dimitar Stanev <jimstanev@gmail.com>
- * @contribution Filip Konstantinos <filip.k@ece.upatras.gr>
+ * @author Dimitar Stanev <jimstanev@gmail.com>, Filip Konstantinos
+ * <filip.k@ece.upatras.gr>
  */
 #pragma once
 
 #include "Exception.h"
-
 #include <algorithm>
+#include <atomic>
 #include <condition_variable>
 #include <functional>
 #include <mutex>
 #include <vector>
 
 namespace OpenSimRT {
+/**
+ * Select mode for data retrieval from the buffer.
+ *
+ * - ON_ENTRY allows the user user to get data from the buffer only after
+ * new data have been added.
+ *
+ * - CONTINUOUS mode allows the user the retrieve data from the buffer
```

Comment:
The continuous is a bit misleading. I would say that you always get the last data. As it is explained, I understand that it gets the value and removes it from the buffer until the buffer is empty. This is not a stack. The name is fine but maybe explain it better.